src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -867,6 +867,8 @@ class G1CollectedHeap : public CollectedHeap {
 
   // Number of regions evacuation failed in the current collection.
   volatile uint _num_regions_failed_evacuation;
+  // Records for every region on the heap whether evacuation failed for it.
+  volatile bool* _regions_failed_evacuation;
```

Comment:
Consider using BitMap and par_xxx operations instead of an array of bool and explicit atomic operations.

Reply:
I would also be fine with the change as is, if that gets this fix for a fairly noisy bug pushed sooner.  So also consider a followup RFE to switch to using BitMap.